lms/djangoapps/grades/models.py

```diff
@@ -257,8 +257,14 @@ def save_grade(cls, **kwargs):
                     usage_key=usage_key,
                     defaults=kwargs,
                 )
+            log.info("persistent subsection grade successfully saved: {0}".format(unicode(grade)))
         except IntegrityError:
             cls.update_grade(user_id=user_id, usage_key=usage_key, **kwargs)
+            log.info("integrity error trying to save grade for user {0}, usage key {1}, defaults {2}".format(
```

Comment:
To loop back on this, I spoke with Eskew, and he told me about the [update_or_create](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#update-or-create) method that exists as well.

So here's a proposal:
```
try:
  with atomic:
    update_or_create
except IntegrityError:
    pass for now, but create a ticket to enqueue a celery task to update the grade
```